lucene/translate.py

```diff
@@ -154,36 +159,35 @@ def postprocess_translated_objc(path):
             if any(fnmatch.fnmatch(full_path_java, ptn) for ptn in excluded):
                 continue
 
-            full_path_m = full_path_java.replace(
-                src, dst).replace(".java", ".m")
+            full_path_m = full_path_java.replace(src, dst).replace(".java", ".m")
             if os.path.exists(full_path_m):
                 if os.path.getmtime(full_path_m) >= os.path.getmtime(full_path_java):  # nopep8
                     continue
             to_compile.append(full_path_java)
-            to_postprocess.append(full_path_m)
 
-    print('%s => %s files to compile' % (src, len(to_compile)))
+    print('\nProcessing: %s, %d java files to compile.\n' % (src, len(to_compile)))
 
-    if to_compile:
-        print('Compiling %d java files' % len(to_compile))
+    for file_to_compile in to_compile:
+        print('Compiling: ' + file_to_compile)
         args = [
-            'j2objc',
-            # '-use-arc',
-            '-classpath', classpaths,
-            '--segmented-headers',
-            '-sourcepath', src,
+            j2objc,
             '-d', dst,
+            '-sourcepath', ' '.join(classpaths),
```

Comment:
There are two problems with this change:

1. `classpaths` are defined as `classpaths = LUCENE_SRC_PATHS + extra_cps`, and since `extra_cps` include compiled class files, the argument should be `-classpath`, not `-sourcepath` (`-classpath` causes `j2objc` – which is modeled after `javac` – to search for both source files and class files).
2. the paths needed to be joined by `:`, not whitespace. That was the cause of your translation failures.